### PR TITLE
base upgrade tests on agoric-3-proposals

### LINKED ISSUE #8443: Separate chain upgrade tests from SDK
## What is the Problem Being Solved?

The [Dockerized Chain Upgrade Tester](https://github.com/Agoric/agoric-sdk/tree/master/packages/deployment/upgrade-test#dockerized-chain-upgrade-tester) is an essential tool for verifying that a particular proposal will succeed when executed on Mainnet ([agoric-3](https://main.agoric.net/network-config)). The SDK branch [release-mainnet1B](https://github.com/Agoric/agoric-sdk/tree/release-mainnet1B/) has been maintained as the current code on Mainnet.

This poses two problems:
1. Any improvements to the _tooling_ (which aren't part of Mainnet) have to be cherry-picked back to that branch in order to be used for testing. And as master drifts from that branch it becomes less feasible.
2. There are proposals on Mainnet that have nothing to do with the Agoric SDK (e.g. [#53
KREAd-rc1 to Mainnet](https://bigdipper.live/agoric/proposals/53)) but should be captured in the upgrade history this test covers

## Description of the Design

Move the "chain upgrade test" concern out of SDK and into its own repo.

Approach
1. Start a new repo in the Agoric org (`mainnet-proposals` because it's testing proposals (ones to come, stacked on ones that have run)
2. Move the `upgrade-test` directory from agoric-sdk over to it. (perhaps using git magic to maintain commit history)
3. Test https://github.com/Agoric/agoric-sdk/issues/8329 upon it
4. Set up its CI to run the tests on every push (must just maser, since any PR in this repo is specifically about the test)

## Security Considerations
n/a
## Scaling Considerations
n/a

## Upgrade Considerations

Proposals to Mainnet should have PRs on this repo before going to stakers.

### PR DESCRIPTION
**Pull request by turadg**
closes: #8443 

## Description

Passed/past upgrades are now tested in https://github.com/Agoric/agoric-3-proposals

This repo need only test the pending upgrade. The a3p tests pending core-eval proposals, but an upgrade proposal entails the whole SDK so it makes sense to test it where that code is.

This PR makes the current upgrade tests start from `ghcr.io/agoric/agoric-3-proposals:main` instead of ag0. That is built by,
- https://github.com/Agoric/agoric-3-proposals/pull/11

### Security Considerations

n/a

### Scaling Considerations

n/a

### Documentation Considerations

Improves visibility of the upgrade proposals and their tests.

### Testing Considerations

Improves CI efficiency. No longer runs the entire upgrade history.
(Before: [29](https://github.com/Agoric/agoric-sdk/actions/runs/6739090244/job/18319951996) to [43](https://github.com/Agoric/agoric-sdk/actions/runs/6739090244/job/18319951901) min , After: maybe [7](https://github.com/Agoric/agoric-sdk/actions/runs/6774366445/job/18411268722), depends on how long the disabled part takes )

### Upgrade Considerations

"

### CODE REVIEW

**Review by raphdev — APPROVED** [review]
Looks good to me. Agree that nightly integration is not a blocker. Linked issue discusses potential approaches for integration.

**Review by mhofman — CHANGES_REQUESTED** [review]
> The one regression is we won't get integration of proposals with each commit to agoric-sdk. That repo's pending proposals use `agoric-sdk:dev` which this repo builds on ever push so it will get those changes whenever it runs CI. That currently is only on PRs and trunk changes. We can later wire up some linking like getting-started has. Or simply have a nightly integration. Neither do I consider to be a blocker for removing the outdated tests here, but I invite the reviews to weigh in on that.

I'm not following, this dependency feels reversed to me. The way I see it, `agoric-3-proposals` would only contain proposals that have passed to the agoric mainnet chain. Then `agoric-sdk` CI would pull an image published from the `agoric-3-proposals` repo, and run a small integration test for every pull request simulating the next chain software upgrade. I am not sure how we would simulate core evals that are not triggered by software upgrades with that setup, but I don't think we need to solve that for now.

In general I do no want to lose the ability to run a simulated chain software upgrade from agoric-sdk CI. I actually need this capability for some upgrade-12 PRs.

File: packages/deployment/upgrade-test/Dockerfile [hunk 1]
```diff
@@ -1,147 +1,18 @@
...
-WORKDIR /usr/src/agoric-sdk/
-COPY ./env_setup.sh ./start_to_to.sh ./upgrade-test-scripts/
-RUN cd upgrade-test-scripts && yarn
-RUN echo '. /usr/src/agoric-sdk/upgrade-test-scripts/env_setup.sh' >> ~/.bashrc
+ENV THIS_NAME= UPGRADE_TO=agoric-upgrade-12 UPGRADE_INFO=${UPGRADE_INFO_12} BOOTSTRAP_MODE=${BOOTSTRAP_MODE}
 
-COPY ./${THIS_NAME} ./upgrade-test-scripts/${THIS_NAME}/
-RUN chmod +x ./upgrade-test-scripts/*.sh
+WORKDIR /usr/src/agoric-sdk/
+COPY --chmod=755 ./env_setup.sh ./start_to_to.sh ./upgrade-test-scripts/
```
Comment (mhofman) [inline packages/deployment/upgrade-test/Dockerfile:15]: Wondering why the support scripts still need to be in this repo? I though we could get away with only having the test files themselves, and this Dockerfile (and Makefile).

Comment (turadg) [reply packages/deployment/upgrade-test/Dockerfile:15]: The logic to upgrade and then wait for upgrade is `start_to_to.sh`. Test files depends on `env_setup.sh`.

There's more cleanup that can be done but that's outside the scope of this PR.

Comment (mhofman) [reply packages/deployment/upgrade-test/Dockerfile:15]: > The logic to upgrade and then wait for upgrade is `start_to_to.sh`. Test files depends on `env_setup.sh`.

But isn't that logic which the base image already needs?

> There's more cleanup that can be done but that's outside the scope of this PR.

Of course, mostly worried about maintenance.

Comment (turadg) [reply packages/deployment/upgrade-test/Dockerfile:15]: > isn't that logic which the base image already needs?

Yes, for now. But I'm trying to minimize coupling. They're already at a different path from here (`/usr/src/upgrade-tests-scripts/` to keep agoric-sdk path clean)

File: packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-12/actions.test.js [hunk 2]
```diff
@@ -27,33 +27,27 @@ test.before(async t => {
   t.context.bundleIds = await installBundles(bundlesData);
 });
 
-test.serial('Open Vaults', async t => {
+test.skip('Open Vaults', async t => {
```
Comment (mhofman) [inline packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-12/actions.test.js:30]: What exactly is failing in this test? It seems to be working on the current versions, no?

**Review by mhofman — APPROVED** [review]
This looks fine to me, but I'd prefer if the vaults test on upgrade-12 was not disabled, as it's the only verification we have that the upgrade didn't create new failures. That said, given the simulated upgrade-12 does not actually change any contract code, I'm ok with it staying failing if needed.

File: packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-12/actions.test.js (hunk 2 above)
Comment (turadg) [reply packages/deployment/upgrade-test/upgrade-test-scripts/agoric-upgrade-12/actions.test.js:30]: Maybe the vaults count but I don't remember for certain.

**Pull request merged**